gateway/modules/eventing/helpers.go

```diff
@@ -52,13 +52,18 @@ func (m *Module) getSpaceCloudIDFromBatchID(batchID string) string {
 	return strings.Split(batchID, "--")[1]
 }
 
-func (m *Module) generateBatchID() string {
-	return fmt.Sprintf("%s--%s", ksuid.New().String(), m.syncMan.GetNodeID())
+func (m *Module) generateBatchID(ID string) string {
+	if ID == "" {
+		ID = ksuid.New().String()
+	}
+	return fmt.Sprintf("%s--%s", ID, m.syncMan.GetNodeID())
 }
 
-func (m *Module) batchRequests(ctx context.Context, requests []*model.QueueEventRequest, batchID string) error {
+func (m *Module) batchRequests(ctx context.Context, eventDocID string, token int, requests []*model.QueueEventRequest, batchID string) error {
```

Comment:
Instead of changing the definition of this function, make another function `batchRequestsRaw` with the same body of the current function. `batchRequests ` will simply call `batchRequestsRaw` with an empty eventDocID and token. This way the functions currently using `batchRequests` will not have to be changed.

You will need to test `batchRequestsRaw`

Reply:
I have already made changes to the functions calling these functions and written tests for them as well

Reply:
And it will create unpredictable fields in the function calling the original function ```batchRequests```

Reply:
Test the raw function